File: javascript/codeeval/214_sort_timestamps.js
var fs  = require("fs");

function st(s) {
    res = []
    ts = s.split(' ')
    for (i = 0; i < ts.length; i++) {
        r = 0
        si = ts[i].split(':')
        r += parseInt(si[0]) * 3600
        r += parseInt(si[1]) * 60
        r += parseInt(si[2])
        res.push([ts[i], r])
    }
    console.log(res.sort(function(a, b) {return a[1] - b[1]}).map(function(x) {return x[0]}).join(' '))
}
fs.readFileSync(process.argv[2]).toString().split('\n').forEach(function (line) {
    if (line != "") {
        st(line)
    }
});

File: ruby/codeeval/214_sort_timestamps.rb
def st(line)
    res = []
    line.split(' ').each do |n|
        r = 0
        si = n.split(':')
        r += si[0].to_i * 3600
        r += si[1].to_i * 60
        r += si[2].to_i
        res << [n, r]
    end
    return res.sort {|a,b| b[1] <=> a[1] }.map{ |x| x[0] }.join(' ')
end
File.open(ARGV[0]).each_line do |line|
    puts(st(line))
end
